Userland/Libraries/LibWeb/HTML/HTMLFormControlsCollection.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2022, Johan Dahlin <jdahlin@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/HTML/HTMLFormControlsCollection.h>
+#include <LibWeb/HTML/Parser/HTMLParser.h>
+
+namespace Web::HTML {
+
+JS::NonnullGCPtr<HTMLFormControlsCollection> HTMLFormControlsCollection::create(DOM::ParentNode& root, Function<bool(DOM::Element const&)> filter)
+{
+    return *root.heap().allocate<HTMLFormControlsCollection>(root.realm(), root, move(filter));
+}
+
+HTMLFormControlsCollection::HTMLFormControlsCollection(DOM::ParentNode& root, Function<bool(DOM::Element const&)> filter)
+    : DOM::HTMLCollection(root, move(filter))
+{
+    set_prototype(&Bindings::cached_web_prototype(realm(), "HTMLFormControlsCollection"));
+}
+
+HTMLFormControlsCollection::~HTMLFormControlsCollection() = default;
+
+// https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#dom-htmlformcontrolscollection-nameditem
+Variant<JS::Handle<RadioNodeList>, JS::Handle<DOM::Element>, Empty> HTMLFormControlsCollection::named_item(FlyString const& name)
+{
+    // 1. If name is the empty string, return null and stop the algorithm.
+    if (name.is_empty())
+        return {};
+
+    // 2. If, at the time the method is called, there is exactly one node in the collection
+    // that has either an id attribute or a name attribute equal to name, then return that node and stop the algorithm.
+    auto elements = root()->children();
+    if (elements->length() == 1) {
```

Comment:
Should this be 1 node total? or 1 node matching the id or name?

Reply:
I read it as exactly one node that also matches id or name and the implementation is supposed to reflect that.

Reply:
Testing this in firefox, it does seem like it means if there is just one element with that name/id. Not only if there is only one element